src/main/java/org/apache/mesos/state/SchedulerState.java

```diff
@@ -0,0 +1,135 @@
+package org.apache.mesos.state;
+
+import com.google.protobuf.TextFormat;
+import org.apache.mesos.Protos.*;
+import org.apache.mesos.curator.CuratorStateStore;
+import org.apache.mesos.offer.TaskException;
+import org.apache.mesos.offer.TaskUtils;
+import org.apache.mesos.reconciliation.TaskStatusProvider;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.*;
+
+/**
+ * Read/write interface for the state of a Scheduler.
+ */
+public class SchedulerState implements TaskStatusProvider {
+    private static final Logger log = LoggerFactory.getLogger(SchedulerState.class);
+    private static final String SUPPRESSED_KEY = "suppressed";
+
+    protected final AbstractStateStore stateStore;
+
+    public SchedulerState(String frameworkName, String mesosZkURI) {
+        this(new CuratorStateStore(frameworkName, mesosZkURI));
+    }
+
+    public SchedulerState(AbstractStateStore stateStore) {
+        this.stateStore = stateStore;
+    }
+
+    public AbstractStateStore getStateStore() {
+        return stateStore;
+    }
+
+    public Optional<FrameworkID> getFrameworkId() {
```

Comment:
Is the added value here the removal of the `throws StateStoreException` from the method signature?  I don't understand why this is needed or wanted.  [The CuratorStateStore in particular already performs the appropriate Exception handling and logging](https://github.com/mesosphere/dcos-commons/blob/master/src/main/java/org/apache/mesos/curator/CuratorStateStore.java#L140-L156).